### DISCUSSION TITLE
AttributeError when importing mesh into preprocessor

### DISCUSSION THREAD

**Discussion by elmeririzarry** [Concrete-Chrono-Development/chrono-preprocessor#16] [Q&A]
With regard to importing CAD or meshes into preprocessor:

I have attempted to generate LDPM model data files using an imported mesh and get the following error (user name redacted):

> 10:34:59  Switching to Chrono Workbench
> 10:34:59  A workbench for building LDPM, CSL, DEM, and SPH models for Project Chrono and other solvers.
> 10:35:07  "Empty widget item in QGridLayout ''."
> 10:38:05  Parameters written to file
> 10:38:05  Moving files.
> 10:38:05  Traceback (most recent call last):
> 10:38:05    File "C:\Users\-\AppData\Roaming\FreeCAD\Mod\chronoConcrete\freecad\chronoWorkbench\modules\mod_LDPMCSL.py", line 159, in generationDriver
> 10:38:05      driver_LDPMCSL(self,fastGen,tempPath)
> 10:38:05    File "C:\Users\-\AppData\Roaming\FreeCAD\Mod\chronoConcrete\freecad\chronoWorkbench\generation\driver_LDPMCSL.py", line 289, in driver_LDPMCSL
> 10:38:05      [meshVertices,meshTets,surfaceNodes,surfaceFaces] = gen_LDPMCSL_initialMesh(cadFile,analysisName,geoName,meshName,minPar)
> 10:38:05    File "C:\Users\-\AppData\Roaming\FreeCAD\Mod\chronoConcrete\freecad\chronoWorkbench\generation\gen_LDPMCSL_initialMesh.py", line 75, in gen_LDPMCSL_initialMesh
> 10:38:05      meshObj.Label = meshName  
> 10:38:05  AttributeError: 'NoneType' object has no attribute 'Label'

For context, this geometry was obtained from a lidar scan, the mesh was simplified using MeshLab, and a volume entity was added using GMSH prior to importing into FreeCAD (in *.vtk format). The following images are (1) the 3D mesh in GMSH, and (2) the result that appears in FreeCAD when attempting to run the LDPM mesh generation. I am aware this same error happens when importing other geometries, not only meshes reconstructed from lidar scans. 

(1)
![image](https://github.com/user-attachments/assets/64f91a1b-88c5-42ac-a7c4-ca3bffc235f2)

(2)
![image](https://github.com/user-attachments/assets/9f38e0f8-aa95-4633-be16-7cde6d4432db)

**Comment by mtroemner** [comment]
Thank you for bringing this to our attention. The bug should now be corrected - please pull the repo and try meshing again.
